### PR TITLE
Clarify when dependency overrides are used

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: src/tools/pub/dependencies.md [hunk 1]
```diff
@@ -396,6 +396,9 @@ dependency_overrides:
   may break your application.
 {{site.alert.end}}
 
+Only dependency overrides from he root pubspec are taken into account
+when doing package resolution.
```
Comment (jonasfj) [inline src/tools/pub/dependencies.md:400]: I think what we want to clearly state is that `dependency_overrides` from the root package is considered.
But `dependency_overrides` inside a package that you depend on will be ignored.

Which means when you publish a package to pub.dev `dependency_overrides` will be ignored, by anyone who uses the package.

Comment (parlough) [reply src/tools/pub/dependencies.md:400]: @sigurdm Any thoughts on the above comments?

I think including this mention is a great idea, Jonas's first sentence particularly makes sense.

Comment (sigurdm) [reply src/tools/pub/dependencies.md:400]: Yeah - I took it as implied - but it is totally worth highlighting!

**Review by domesticmouse — APPROVED** [review]

File: src/tools/pub/dependencies.md [hunk 2]
```diff
@@ -396,6 +396,13 @@ dependency_overrides:
   may break your application.
 {{site.alert.end}}
 
+Only dependency overrides from he root pubspec are considered
```
Comment (parlough) [inline src/tools/pub/dependencies.md:399]:
```suggestion
Only dependency overrides from the root pubspec are considered
```

File: src/tools/pub/dependencies.md [hunk 3]
```diff
@@ -396,6 +396,13 @@ dependency_overrides:
   may break your application.
 {{site.alert.end}}
 
+Only dependency overrides from the root pubspec are considered
+when doing package resolution. This means that dependency overrides
+inside a package that you depend on will be ignored.
+
+Therefore when you publish a package to pub.dev, dependency overrides 
+will be ignored by anyone depending on that package.
```
Comment (parlough) [inline src/tools/pub/dependencies.md:404]:
```suggestion
Only dependency overrides from the root pubspec are considered
when doing package resolution.
This means that dependency overrides
inside a package that you depend on will be ignored.

Therefore, when you publish a package to pub.dev,
dependency overrides will be ignored by anyone depending on that package.
```

**Review by parlough — APPROVED** [review]
This clarification looks good to me and I can see it preventing some confusion. Thanks!

**Review by kwalrath — APPROVED** [review]

**Comment by parlough** [comment]
Looks great!

**Pull request merged**